### PR TITLE
Generate Maven Java 8 projects with correct pom contents

### LINKED ISSUE #2215: maven.compiler.source is 1.7 in pom.xml even when project is set to 1.8
```
    <maven.compiler.source>1.7</maven.compiler.source>
    <maven.compiler.target>1.7</maven.compiler.target>
```

see google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.util/templates/appengine/pom.xml.standard.ftl

and related issue is that it has the servlet API 2.5 instead of 3.1:

```
    <dependency>
      <groupId>javax.servlet</groupId>
      <artifactId>servlet-api</artifactId>
      <version>2.5</version>
      <scope>provided</scope>
    </dependency>
```

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #2215

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java [hunk 1]
```diff
@@ -46,15 +48,20 @@
...
+      String location, String javaPackage, AppEngineRuntime runtime) {
+    return createWebAppProject(bot, projectName, location, javaPackage, runtime,
+        null /* extraBotActions */);
   }
 
-  /** Create a new project with the Maven-based Google App Engine Standard Java Project wizard */
+  /**
+   * Create a new project with the Maven-based Google App Engine Standard Java Project wizard
+   * 
+   * @param runtime
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java]: document or delete

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java [hunk 2]
```diff
@@ -46,15 +48,20 @@
...
+   */
   public static IProject createMavenWebAppProject(final SWTWorkbenchBot bot, String projectName,
-      String location, String javaPackage,
-      final String mavenGroupId, final String mavenArtifactId) {
-    return createWebAppProject(bot, projectName, location, javaPackage, new Runnable() {
+      String location, String javaPackage, AppEngineRuntime runtime, final String mavenGroupId,
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java]: just groupId and artifactId are probably sufficient

**Comment by chanseokoh** [comment]
Tests fail on Mars because the JDK toolchain for Mars is Java 7.

```
testWithPackage_java8(com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest)  Time elapsed: 20.536 sec  <<< FAILURE!
java.lang.AssertionError: New native project has errors
/appWithPackage:-1: Unbound classpath container: 'JRE System Library [JavaSE-1.8]' in project 'appWithPackage'
/appWithPackage:-1: The project cannot be built until build path errors are resolved
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:162)
	at com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest.createAndCheck(NewNativeAppEngineStandardProjectTest.java:106)
	at com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest.testWithPackage_java8(NewNativeAppEngineStandardProjectTest.java:69)


testHelloWorld_java8(com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest)  Time elapsed: 9.007 sec  <<< FAILURE!
java.lang.AssertionError: New Maven project has errors
/appWithPackageProject:-1: The project cannot be built until build path errors are resolved
/appWithPackageProject:-1: Unbound classpath container: 'JRE System Library [JavaSE-1.8]' in project 'appWithPackageProject'
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:162)
	at com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest.createAndCheck(NewMavenBasedAppEngineProjectWizardTest.java:138)
	at com.google.cloud.tools.eclipse.integration.appengine.NewMavenBasedAppEngineProjectWizardTest.testHelloWorld_java8(NewMavenBasedAppEngineProjectWizardTest.java:76)
```

**Comment by briandealwis** [comment]
This test is failing on Mars as it requires a Java 8 JRE (: facepalm:). It needs an `assumeThat` for a Java 8 JRE.
```
testWithPackage_java8(com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest) Time elapsed: 20.536 sec <<< FAILURE!
java.lang.AssertionError: New native project has errors
/appWithPackage:-1: Unbound classpath container: 'JRE System Library [JavaSE-1.8]' in project 'appWithPackage'
/appWithPackage:-1: The project cannot be built until build path errors are resolved
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:162)
	at com.google.cloud.tools.eclipse.integration.appengine.NewNativeAppEngineStandardProjectTest.createAndCheck(NewNativeAppEngineStandardProjectTest.java:106)
```

File: plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java (hunk 2 above)
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/SwtBotAppEngineActions.java]: This is used by existing tests; I've just added an `AppEngineRuntime` parameter.

**Pull request merged**